app/views/courses/TeacherClassView.coffee

```diff
@@ -332,19 +333,21 @@ module.exports = class TeacherClassView extends RootView
     # TODO: Does not yield .csv download on Safari, and instead opens a new tab with the .csv contents
     window.tracker?.trackEvent 'Teachers Class Export CSV', category: 'Teachers', classroomID: @classroom.id, ['Mixpanel']
     courseLabels = ""
-    courseOrder = []
-    courses = (@courses.get(c._id) for c in @classroom.get('courses'))
-    courseLabelsArray = helper.courseLabelsArray courses
+    courses = (@courses.get(c._id) for c in @sortedCourses)
+    courseLabelsArray = helper.courseLabelsArray(courses)
```

Comment:
What we've been doing to cover this case (but which is still less bug-proof) is to do things like recalculate @sortedCourses changes: 
```
  @listenTo @classroom, 'sync change update', =>
    @sortedCourses = @classroom.getSortedCourses()
```